sphinx/ext/autosummary/generate.py

```diff
@@ -221,12 +217,13 @@ def find_autosummary_in_files(filenames):
     """
     documented = []
     for filename in filenames:
-        f = open(filename, 'r')
-        lines = f.read().splitlines()
-        documented.extend(find_autosummary_in_lines(lines, filename=filename))
-        f.close()
+        with open(filename, 'r', encoding='utf-8') as f:
```

Comment:
Since these are .rst sources, we should use the source suffix defined in conf.py. But the config is not available...

Since the purpose here is to find autosummary/module directives, utf-8 with an error handler should be good enough, except if people use utf-16 or other non-ascii compatible encodings.

But in any case using open() with encoding kwarg here is not compatible with Python 2.